src/cloudai/configurator/env_params.py

```
@@ -0,0 +1,212 @@
+
+    @model_validator(mode="after")
+    def _validate_weights(self) -> Self:
+        if self.weights is None:
+            return self
+        if len(self.weights) != len(self.values):
+            raise ValueError(
+                f"env_params weights length {len(self.weights)} does not match values length {len(self.values)}"
+            )
+        for w in self.weights:
+            if w < 0:
+                raise ValueError(f"env_params weights must be non-negative; got {w}")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)